### PR TITLE
EASY-2316: Create Transformer to add organizationId

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
fixes EASY-2316

#### When applied it will
* Create a Transformer rule to add organizationId to the specified organisations

#### Where should the reviewer @DANS-KNAW/easy start?

### CODE REVIEW

File: src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        </emd:creator>
+      </emd:easymetadata>
+
+    Transformer("EMD", "orgISNI", "BAAC bv", "0000 0004 7237 0000")
+      .transform(inputXML).headOption.map(new PrettyPrinter(160, 2).format(_))
+      .value shouldBe new PrettyPrinter(160, 2).format(expectedXML)
+  }
+
+  it should "add ROR organizationId" in {
+
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala]:
```suggestion
```

File: src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.umd
+
+import org.scalatest.{FlatSpec, Inside, Matchers, OptionValues}
+
+import scala.xml.PrettyPrinter
+
+class IsniTransformerSpec extends FlatSpec with Matchers with OptionValues with Inside {
+
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala:38]:
```suggestion
```

File: src/main/scala/nl.knaw.dans.easy.umd/Transformer.scala [hunk 3]
```diff
@@ -69,6 +71,17 @@ object Transformer {
...
 
+  private def organisationIdTransformer(organisationName: String, organisationText: String, organisationURI: String, scheme: String): RuleTransformer = {
+    new RuleTransformer(new RewriteRule {
+      override def transform(n: Node): Seq[Node] = n match {
+        case e:Elem if e.label.equals("organization") && e.text.equals(organisationName) =>
+          e ++ <eas:organizationId eas:identification-system={organisationURI} eas:scheme={scheme}>{scheme}: {organisationText}</eas:organizationId>
+        case other => other
+      }
+    })
+
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.umd/Transformer.scala]:
```suggestion
```

File: src/main/scala/nl.knaw.dans.easy.umd/Transformer.scala [hunk 4]
```diff
@@ -69,6 +71,17 @@ object Transformer {
     })
   }
 
+  private def organisationIdTransformer(organisationName: String, organisationText: String, organisationURI: String, scheme: String): RuleTransformer = {
+    new RuleTransformer(new RewriteRule {
+      override def transform(n: Node): Seq[Node] = n match {
+        case e:Elem if e.label.equals("organization") && e.text.equals(organisationName) =>
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.umd/Transformer.scala]:
```suggestion
        case e:Elem if e.label == "organization" && e.text == organisationName =>
```

**Review by rvanheest — COMMENTED** [review]
Changes look good! just some comments on empty lines and Scala syntax/conventions.
Please also run the formatter on these two files!

File: src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            <eas:surname>B</eas:surname>
+            <eas:organization>BAAC</eas:organization>
+            <eas:entityId eas:scheme="DAI"></eas:entityId>
+          </eas:creator>
+          <eas:creator>
+            <eas:title>drs</eas:title>
+            <eas:initials>MC</eas:initials>
+            <eas:surname>R</eas:surname>
+            <eas:organization>BAAC bv</eas:organization>
+            <eas:organizationId eas:identification-system="http://isni.org/isni/0000000472370000" eas:scheme="ISNI">ISNI: 0000 0004 7237 0000</eas:organizationId>
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala]: A "what's in a name" question:

What has an id-**system** to do with and id-**instance**? Shouldn't this become something like:
```suggestion
            <eas:organizationId eas:identification-system="http://isni.org/isni/" eas:scheme="ISNI">ISNI: 0000 0004 7237 0000</eas:organizationId>
```

Comment (lindareijnhoudt) [reply src/test/scala/nl.knaw.dans.easy.umd/IsniTransformerSpec.scala]: you're right of course 👍

**Review by rvanheest — APPROVED** [review]

**Pull request merged**